### PR TITLE
Migrate all build.gradle to version catalog and use plugin block.

### PR DESCRIPTION
**Pull request by devPalacio**
* Migrated dependency versions to the version catalog.
* Added the dependency guard plugin https://github.com/dropbox/dependency-guard
* Using modern Gradle Plugin block syntax.

### CODE REVIEW

File: ChangeLog.md [hunk 1]
```diff
@@ -1,3 +1,15 @@
+## 5.4.0 (2022-09-20)
```
Comment (handstandsam) [inline ChangeLog.md:1]: We will need to update, but this is a good placeholder.

File: ChangeLog.md [hunk 2]
```diff
@@ -1,3 +1,15 @@
...
+---------------------------------------------
+- Update dropbox-api-spec to point to more recent version (Sept 01, 2022) [#431](https://github.com/dropbox/dropbox-sdk-java/pull/431)
+- Generated stone api code is now checked into repository for greater visibility of spec changes [#418](https://github.com/dropbox/dropbox-sdk-java/pull/418)
+- Changed from `Master` -> `Main` [#424](https://github.com/dropbox/dropbox-sdk-java/pull/424)
```
Comment (handstandsam) [inline ChangeLog.md:5]: `master` -> `main`

File: ReadMe.md [hunk 3]
```diff
@@ -24,7 +24,7 @@ If you're using Maven, then edit your project's "pom.xml" and add this to the `<
 <dependency>
     <groupId>com.dropbox.core</groupId>
     <artifactId>dropbox-core-sdk</artifactId>
-    <version>5.3.0</version>
+    <version>5.4.0</version>
```
Comment (handstandsam) [inline ReadMe.md:27]: May want to keep the docs at 5.3.0 until we release.

File: ReadMe.md [hunk 4]
```diff
@@ -33,15 +33,15 @@ If you are using Gradle, then edit your project's "build.gradle" and add this to
 ```groovy
 dependencies {
     // ...
-    implementation 'com.dropbox.core:dropbox-core-sdk:5.3.0'
+    implementation 'com.dropbox.core:dropbox-core-sdk:5.4.0'
```
Comment (handstandsam) [inline ReadMe.md:36]: same

File: dropbox-sdk-android/build.gradle [hunk 5]
```diff
@@ -1,9 +1,10 @@
 plugins {
     id 'org.jetbrains.kotlinx.binary-compatibility-validator' version '0.11.1'
+    alias(dropboxJavaSdkLibs.plugins.maven.publish.plugin) apply false
+    alias(dropboxJavaSdkLibs.plugins.gradle.version.plugin) apply false
```
Comment (handstandsam) [inline dropbox-sdk-android/build.gradle:4]: Should these go in root build.gradle?

File: examples/android/build.gradle [hunk 6]
```diff
@@ -1,5 +1,7 @@
-apply plugin: 'com.android.application'
-apply plugin: 'kotlin-android'
+plugins {
+    id 'com.android.application'
+    id 'kotlin-android'
```
Comment (handstandsam) [inline examples/android/build.gradle:3]: `id "org.jetbrains.kotlin.android"` or use the `alias(versioncataloggoeshere)`

File: ChangeLog.md (hunk 1 above)
Comment (devPalacio) [reply ChangeLog.md]: I moved all the 5.4 references to a separate PR

File: ChangeLog.md (hunk 2 above)
Comment (devPalacio) [reply ChangeLog.md]: I moved all the 5.4 references to a separate PR

File: dropbox-sdk-android/build.gradle (hunk 5 above)
Comment (devPalacio) [reply dropbox-sdk-android/build.gradle]: Good call!

File: examples/android/build.gradle (hunk 6 above)
Comment (devPalacio) [reply examples/android/build.gradle]: done!

**Pull request merged**